File: TextEditor/TextEditor/TextEditor/EditorMemento.cs
using System.Runtime.Serialization;

namespace TextEditor;

public class EditorMemento
{
    private readonly string text;
    private readonly int cursorPosition;
    private readonly string selection;

    public EditorMemento(string text, int cursorPosition, string selection)
    {
        this.text = text;
        this.cursorPosition = cursorPosition;
        this.selection = selection;
    }

    public string GetText()
    {
        return text;
    }

    public int GetCursorPosition()
    {
        return cursorPosition;
    }

    public string GetSelection()
    {
        return selection;
    }
    
}

File: DPLab12_Memeto/DPLab09_mediator/src/EditorMemento.java
public class EditorMemento {
    private String content;
    private int cursorPosition;
    private String selection;

    public EditorMemento(String content, int cursorPosition, String selection) {
        this.content = content;
        this.cursorPosition = cursorPosition;
        this.selection = selection;
    }

    public String getContent() {
        return content;
    }

    public int getCursorPosition() {
        return cursorPosition;
    }

    public String getSelection() {
        return selection;
    }
}
